### PR TITLE
xkcd

### PR DESCRIPTION
**Pull request by nikhiljha**
I wrote 20 short programs in Python yesterday.  It was wonderful.  Perl, I'm leaving you.

### CODE REVIEW

File: ircbot/plugin/xkcd.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+def register(bot):
+    bot.listen(REGEX.pattern, show_comic)
+
+
+def show_comic(_, msg):
+    """Show XKCD comic details."""
+    for number in REGEX.findall(msg.text):
+        try:
+            comic = xkcd.Comic(int(number))
+            reply = f"XKCD#{int(number)} | '{comic.getTitle()}' | {comic.getImageLink()}"
```
Comment (dkess) [inline ircbot/plugin/xkcd.py:18]: Link to the comic instead of the image IMO

Comment (cg505) [reply ircbot/plugin/xkcd.py:18]: linking the image directly will cause it to expand in matrix/slack. this message might be responding to a link anyway

Comment (jvperrin) [reply ircbot/plugin/xkcd.py:18]: You definitely lose the hover/alt text when just having the image, which detracts from the comic. That could be posted in a second line or something but I do think it needs to be either included or the full comic linked to.

File: ircbot/plugin/xkcd.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            msg.respond(str(reply))
```
Comment (ja5087) [inline ircbot/plugin/xkcd.py:19]: fstrings are already strings?

File: ircbot/plugin/xkcd.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        except AssertionError:
```
Comment (ja5087) [inline ircbot/plugin/xkcd.py:20]: Where is this meant to catch?

File: ircbot/plugin/xkcd.py (hunk 1 above)
Comment (nikhiljha) [reply ircbot/plugin/xkcd.py:18]: I was considering linking to explainxkcd too, that has everything + an explanation if you need it :P

Comment (ja5087) [reply ircbot/plugin/xkcd.py:18]: I think just post the image and alt text. The point is to present the comic inline within the conversation, not so we have to click another link.

Comment (cg505) [reply ircbot/plugin/xkcd.py:18]: yeah, alt text is important

File: ircbot/plugin/xkcd.py (hunk 3 above)
Comment (nikhiljha) [reply ircbot/plugin/xkcd.py]: oops, was supposed to check `HTTPError` in case it's not a valid comic (and therefore 404s)

**Comment by nikhiljha** [comment]
Ok, tested and should work now 👍

**Pull request merged**